frontend/packages/noobaa-storage-plugin/src/components/bucket-class/backingstore-table.tsx

```diff
@@ -0,0 +1,279 @@
+import { getNamespace, getName, useSelectList, getUID } from '@console/shared';
+import { Table, ListPage, getFilteredRows } from '@console/internal/components/factory';
+import { ResourceLink } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { getBucketName, getRegion, getBackingStoreType, getBSLabel } from '../../utils';
+import CreateBackingStoreFormModal from '../create-backingstore-page/create-bs-modal';
+import { NooBaaBackingStoreModel } from '../../models';
+import { BackingStoreKind, PlacementPolicy } from '../../types';
+import './_backingstore-table.scss';
+
+const tableColumnClasses = [
+  '',
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+  classNames('pf-m-hidden', 'pf-m-visible-on-sm'),
+];
+
+const getColumns = () => [
+  {
+    title: 'Name',
+    sortField: 'metadata.name',
+    transforms: [sortable],
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'Bucket Name',
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Type',
+    props: { className: tableColumnClasses[2] },
+  },
+  {
+    title: 'Region',
+    props: { className: tableColumnClasses[3] },
+  },
+];
+
+const getRows: GetRows = (rowProps, selectedItems) => {
+  const {
+    componentProps: { data },
+  } = rowProps;
+
+  const rows = data.map((bs) => {
+    const cells: IRow['cells'] = [
+      {
+        title: (
+          <ResourceLink
+            linkTo={false}
+            kind={referenceForModel(NooBaaBackingStoreModel)}
+            name={getName(bs)}
+            namespace={getNamespace(bs)}
+          />
+        ),
+      },
+      {
+        title: getBucketName(bs) || '-',
+      },
+      {
+        title: getBackingStoreType(bs) || '-',
+      },
+      {
+        title: getRegion(bs) || '-',
+      },
+    ];
+    return {
+      cells,
+      selected: selectedItems?.has(bs.metadata.uid),
+      props: {
+        id: getUID(bs),
+      },
+    };
+  });
+  return rows;
+};
+
+const BackingStoreTable: React.FC<BackingStoreTableProps> = (props) => {
+  const {
+    customData: { onRowsSelected, preSelected },
+    data,
+    filters = {},
+  } = props;
+  const visibleRows = getFilteredRows(filters, null, data);
+  const visibleUIDs = React.useMemo(() => new Set<string>(visibleRows?.map(getUID)), [visibleRows]);
+  const { onSelect, selectedRows, updateSelectedRows } = useSelectList<BackingStoreKind>(
+    data,
+    visibleUIDs,
+    onRowsSelected,
+  );
+
+  React.useEffect(() => {
+    if (!_.isEmpty(preSelected) && selectedRows.size === 0) {
+      const preSelectedRows = data.filter((item) => preSelected.includes(getUID(item)));
+      updateSelectedRows(preSelectedRows);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [JSON.stringify(preSelected), selectedRows.size, updateSelectedRows, JSON.stringify(data)]);
```

Comment:
you could use `useDeepCompareMemoize(preSelected, true)` which will memoizes the value using `JSON.stringify` and use the memoized value for useEffect dependency array. That way you wont have to disable the eslint rule and will be safer in the long run. (same for `data`)